### DISCUSSION TITLE
x-trap does not working in Shadow DOM

### DISCUSSION THREAD

**Discussion by bty888** [alpinejs/alpine#3663] [7. Bugs (archived)]
hi ,
i'm trying to implement this example show in the documentation  https://alpinejs.dev/plugins/focus#x-trap using web components (with Shadow DOM) , the tabbing dose not work with shadow DOM but does without it .

A snippet to reproduce this issue : https://codepen.io/vm09/pen/vYQxBqY

**Comment by ekwoka** [comment]
The internals for working with tababbles doesn't pierce the shadow dom. The shadow dom is for internals to the component that other things aren't supposed to go into.

**Reply by bty888** [reply]
shadow DOM  support is added by focus-trap in 6.8.0 , seems like @alpine/focus has a build issue when publishing .

**Comment by bty888** [comment]
@calebporzio  looks like the build for @alpine/focus has an issue with resolving focus-trap dependency , the published version has focus-trap version of  6.6.1 instead of 6.9.4 .

**Reply by bty888** [reply]
@calebporzio this maybe due to .npm cache on the system , resolving this publishing issue would greatly appreciative ,thank you.

**Reply by ekwoka** [reply]
It's not that. The dependency is 6.6.1, not 6.9.4.

It just isn't updated.

Good to know that was added :)

**Reply by bty888** [reply]
yeah, but the dependence for focus trap is  ^6.6.1 
 (https://github.com/alpinejs/alpine/blob/main/packages/focus/package.json#L17) 
which will update you to all future minor/patch versions on installing dependences
i see the package-lock has the updated version too ( 6.9.4) but somehow in the CI build (works fine locally) is unable to fetch update to the Package .
https://github.com/alpinejs/alpine/blob/main/package-lock.json#L3351

**Reply by ekwoka** [reply]
That only applies when the package is a dependency.

Not when the package is directly installing it's own dependencies.

So when you add focus as a dep, you'll see it installs the latest version of focus-trap alongside focus.

but focus bundles it's own dependency, which is the version stated in the package.json

broadly speaking, the whole build system for alpine is FUBAR. So it will likely need to be directly bumped up on the package.json directly, and should if Alpine wants to expressly support this feature.

**Reply by bty888** [reply]
@ekwoka thank you for the insight on the build system , i will open a PR to address this.

**Reply by bty888** [reply]
https://github.com/alpinejs/alpine/pull/3667 is created to address this issue

**Reply by biswaoecetc** [reply]
Thank you for the fix @bty888. We had this issue for the shadow DOM implementation. It finally working with this fix. Saved my days
